include/eld/Config/GeneralOptions.h

```diff
@@ -834,6 +837,16 @@ class GeneralOptions {
   void setTimingStatsFile(std::string StatsFile) {
     TimingStatsFile = StatsFile;
   }
+  //--------------------Memory Report--------------------------------
+  // --print-memory-report
+  bool showMemoryReport() const { return PrintMemoryReport; }
+
+  void setPrintMemoryReport() { PrintMemoryReport = true; }
+
+  // --emit-memory-report <file>
+  std::string getMemoryReportFile() const { return MemoryReportFile; }
+
+  void setMemoryReportFile(std::string F) { MemoryReportFile = F; }
```

Comment:
I think we should try to use `llvm::StringRef` / `std::string_view` wherever we can as they avoid copying strings. What do you think about it?

```suggestion
  void setMemoryReportFile(llvm::StringRef F) { MemoryReportFile = F; }
```